src/compiler/checker.ts

```diff
@@ -30752,7 +30752,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             // If no inferences have been made, and none of the type parameters for which we are inferring
             // specify default types, nothing is gained from instantiating as type parameters would just be
             // replaced with their constraints similar to the apparent type.
-            if (inferenceContext && contextFlags! & ContextFlags.Signature && some(inferenceContext.inferences, hasInferenceCandidatesOrDefault)) {
+            if (
+                inferenceContext &&
+                (contextFlags! & ContextFlags.Signature || maybeTypeOfKind(contextualType, TypeFlags.Simplifiable)) &&
+                some(inferenceContext.inferences, hasInferenceCandidatesOrDefault)
+            ) {

```

Comment:
It turns out that this breaks **one** new test case (merged quite recently and related to a recent change, see change [here](https://github.com/microsoft/TypeScript/pull/48838/files#diff-9ca395ead99032851b41b9bc0b00747000504fa87686bb95acab928cf99fb75b)). It can be fixed with:
```diff
diff --git a/src/compiler/checker.ts b/src/compiler/checker.ts
index 5b1cf22c29..b3f719a1ca 100644
--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -30759,7 +30759,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             ) {
                 // For contextual signatures we incorporate all inferences made so far, e.g. from return
                 // types as well as arguments to the left in a function call.
-                return instantiateInstantiableTypes(contextualType, inferenceContext.nonFixingMapper);
+                const instantiated = instantiateInstantiableTypes(contextualType, inferenceContext.nonFixingMapper);
+                if (contextFlags! & ContextFlags.Signature) {
+                    return instantiated;
+                }
+                return getIntersectionType([contextualType, instantiated]);
             }
             if (inferenceContext?.returnMapper) {
                 // For other purposes (e.g. determining whether to produce literal types) we only
```

However, I kinda feel that this particular potential fix would be better put into https://github.com/microsoft/TypeScript/pull/56953 

it's related to indexed access and my very initial goal behind this PR here 2 years ago was to introduce this change for conditional types. Only later on I realized that both belong to the `Simplifiable` group and I extended this PR. So one possible course of action would also be to revert support for `IndexedAccess` here for now and revisit this later.

I also don't think that this particular broken test case is that important. It was kind of created artificially~ by @jfet97 and me. It was just an attempt to make certain things work - and not necessarily a way that we wanted to write types with. That doesn't exactly mean that this is bad per se but perhaps it doesn't have to be fixed immediately. It didn't work in TS 5.3 **anyway** (see this [TS playground](https://www.typescriptlang.org/play?ts=5.3.3#code/C4TwDgpgBAysCGwIGED2A7AZgSwOYB4AVAQQGNhsMoIAPJdAEwGcongAnbdXAPigF4oAbwBQUaug4gA-AC4oJcpXRjWCJEzlQAShFKp2DfG07cANLHUoMOAoooYePANwiAvq5GhIlxBBikABYQALbwAsKqbH6a8rr6hsYcXLgWcH4BwWEu7p4MegA28OzQmACu6EpUpCV+ALLwQVwQ+KqEaFh41HQQjCzpSB22RGQO6DxmbaPK3fTMaqa4Ee02eADaAES9UhsAurO98yYpUNIKQ+tbkuwge1Dyx+YiPAAU+p248kJQawDSUFwoABrCAgVCYc6rJYAMmBoPBvkGUPw8HQIB4u3kKw+f32bgAlPI1vZlBZsbZdp53mwAVgIOwSgwAArFeAhJgARgiNQg9UagWaL1E4m2N3kG0wqFQG0m4miGi+qnE8EV4jVEik4oARsUZUqoG5ZQajbQOCqoByAEyTAlUjA0riYemMlnsNlMS3c2pIBpNdAQIWqUUgcWS6VG+UQJiqtXm4XqjViqASqV69WG1QZkV0N3yK02-HOIA) where the reverse mapped types are **not** inferred).

This patch from the above breaks only one case, the one introduced in https://github.com/microsoft/TypeScript/pull/53647 . The break is that it stops infer tuples so perhaps I will be able to find a simple fix later for it by adjusting `inTupleContext`.

In the meantime though, I'd love to learn what impact this PR has on the community tests etc in its current form.

Reply:
The broken test case from #55811 could be easily fixed by using a `const` generic or with an `as const` assertion, therefore it doesn't seem like a big deal to me either.